spark-connect/server/src/test/scala-spark-master/io/delta/connect/DeltaConnectPlannerSuite.scala

```diff
@@ -102,4 +113,256 @@ class DeltaConnectPlannerSuite
+    withTempDir { dir =>
+      spark.range(100)
+        .select(col("id") % 10 as "part", col("id") as "value")
+        .write
+        .partitionBy("part")
+        .format("parquet")
+        .mode("overwrite")
+        .save(dir.getAbsolutePath)
+
+      val input = createSparkRelation(
+        proto.DeltaRelation
+          .newBuilder()
+          .setConvertToDelta(
+            proto.ConvertToDelta
+              .newBuilder()
+              .setIdentifier(s"parquet.`${dir.getAbsolutePath}`")
+              .setPartitionSchemaString("part LONG")
+          )
+      )
+      val plan = transform(input)
+      val result = Dataset.ofRows(spark, plan).collect()
+
+      assert(result.length === 1)
+      val deltaTable = DeltaTable.forName(spark, result.head.getString(0))
+      assert(!deltaTable.toDF.isEmpty)
+    }
+  }
+
+  test("convert to delta with partitioning schema struct") {
+    withTempDir { dir =>
+      spark.range(100)
+        .select(col("id") % 10 as "part", col("id") as "value")
+        .write
+        .partitionBy("part")
+        .format("parquet")
+        .mode("overwrite")
+        .save(dir.getAbsolutePath)
+
+      val input = createSparkRelation(
+        proto.DeltaRelation
+          .newBuilder()
+          .setConvertToDelta(
+            proto.ConvertToDelta
+              .newBuilder()
+              .setIdentifier(s"parquet.`${dir.getAbsolutePath}`")
+              .setPartitionSchemaStruct(
+                spark_proto.DataType
+                  .newBuilder()
+                  .setStruct(
+                    spark_proto.DataType.Struct
+                      .newBuilder()
+                      .addFields(
+                        spark_proto.DataType.StructField
+                          .newBuilder()
+                          .setName("part")
+                          .setNullable(false)
+                          .setDataType(
+                            spark_proto.DataType
+                              .newBuilder()
+                              .setLong(spark_proto.DataType.Long.newBuilder())
+                          )
+                      )
+                  )
+              )
+          )
+      )
+      val plan = transform(input)
+      val result = Dataset.ofRows(spark, plan).collect()
+
+      assert(result.length === 1)
+      val deltaTable = DeltaTable.forName(spark, result.head.getString(0))
+      assert(!deltaTable.toDF.isEmpty)
+    }
+  }
+
+  test("history") {
+    withTable("table") {
+      DeltaTable.create(spark).tableName(identifier = "table").execute()
+
+      val input = createSparkRelation(
+        proto.DeltaRelation
+          .newBuilder()
+          .setDescribeHistory(
+            proto.DescribeHistory.newBuilder()
+              .setTable(proto.DeltaTable.newBuilder().setTableOrViewName("table"))
+          )
+      )
+
+      val result = new SparkConnectPlanner(createDummySessionHolder(spark)).transformRelation(input)
```

Comment:
@andreaschat-db It's the right API to be used in this context - https://github.com/apache/spark/blob/02795a36171cac51b1beaf68c42e8dfe3e6f0903/sql/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala#L135, we don't have `transform`.

Reply:
Alright but what is the difference with the test above where you use `val plan = transform(input)` on a `spark_proto.Relation`?

Reply:
@andreaschat-db You are right, great catch, I was debugging that test because of mismatch result and replaced `transform` with `transformRelation`, but forgot to put it back, let's use `transform` for unity.